src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs

```diff
@@ -197,6 +211,24 @@ public void CopyTo(T[] array, int arrayIndex, int count)
         /// <returns>The position in this builder that was copied up to.</returns>
         public CopyPosition CopyTo(CopyPosition position, T[] array, int arrayIndex, int count)
         {
+            int CopyToCore(T[] sourceBuffer, int sourceIndex)
+            {
+                Debug.Assert(sourceBuffer.Length > sourceIndex);
```

Comment:
@jamesqko it seems that this is false sometimes, even with simple cases like: `new int[] { 1 }.Concat(new int[] { 2 }).ToArray()`, these two are equal. I believe the problem is having no elements to copy 
(count: 0). The function should simply return.